### PR TITLE
[RNMobile] Add Reusable blocks to the inserter menu

### PR DESCRIPTION
**Pull request by fluiddot**
`gutenberg-mobile` PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/3054

## Description

Add support to insert existing Reusable blocks from the editor in the mobile version. For this purpose the inserter menu now shows two tabs in case this kind of blocks have been previously created in the site.

The Reusable tab will show a list of all created Reusable blocks of the site.

This PR also includes fixes for the following issues:
- https://github.com/wordpress-mobile/gutenberg-mobile/issues/2689

### ⚠️ Caveats

- Until we add support for [unbounded queries](https://github.com/wordpress-mobile/gutenberg-mobile/issues/2661), the app only fetches the last 100 reusable blocks from the site so the inserter menu will show up to 100 items.
- The search form takes into account the reusable blocks but I'd like to note that the feature is still WIP so it could change in the future.
_**EDIT:** The search form is only available in development mode for now._

## How has this been tested?

I tested this feature in the three different scenarios:

<details>
<summary><strong>WordPress.com site</strong></summary>

Create a WordPress.com site or use an already created one.
1. Create a post in the web version
1. Add some blocks and create a Reusable block
1. Open the app with metro running (Gutenberg-mobile)
1. Add the WordPress.com site
1. Create/edit a post/page
1. Tap on the ➕  button to display the inserter menu
1. Observe that 2 tabs are shown, one (first one to be presented) with the blocks and another one with reusable blocks
1. Observe that there're no reusable blocks in the blocks tab
1. Navigate to the reusable tab
1. Observe that previously created reusable blocks are shown
1. Tap on a reusable block
1. Observe that the block has been inserted into the post and that is in the proper position

</details>

<details>
<summary><strong>Self-hosted site with Jetpack</strong></summary>

Create a self-hosted site with Jetpack (I used https://jurassic.ninja/create/) or use an already created one.
1. Create a post in the web version
1. Add some blocks and create a Reusable block
1. Open the app with metro running (Gutenberg-mobile)
1. Add the self-hosted site
1. Create/edit a post/page
1. Tap on the ➕  button to display the inserter menu
1. Observe that 2 tabs are shown, one (first one to be presented) with the blocks and another one with reusable blocks
1. Observe that there're no reusable blocks in the blocks tab
1. Navigate to the reusable tab
1. Observe that previously created reusable blocks are shown
1. Tap on a reusable block
1. Observe that the block has been inserted into the post and that is in the proper position

</details>

<details>
<summary><strong>Self-hosted site without Jetpack</strong></summary>

Create a self-hosted site without Jetpack (I created a local one using [this instructions](https://developer.wordpress.org/block-editor/contributors/develop/getting-started/#local-environment)) or use your own.
1. Create a post in the web version
1. Add some blocks and create a Reusable block
1. Open the app with metro running (Gutenberg-mobile)
1. Add the self-hosted site
1. Create/edit a post/page
1. Tap on the ➕  button to display the inserter menu
1. Observe that 2 tabs are shown, one (first one to be presented) with the blocks and another one with reusable blocks
1. Observe that there're no reusable blocks in the blocks tab
1. Navigate to the reusable tab
1. Observe that previously created reusable blocks are shown
1. Tap on a reusable block
1. Observe that the block has been inserted into the post and that is in the proper position

</details>

### Additional tests to prevent regressions

It's required to have some reusable blocks already created for the following additional tests.

<details><summary><strong>Clipboard</strong></summary>

1. Copy a block
2. Observe that the copied block is the first item when opening the inserter menu
3. Navigate to the reusable tab
4. Observe that it doesn't show the copied block

</details>

<details><summary><strong>Insert block</strong></summary>

1. Tap on the ➕ button to display the inserter menu
2. Tap on a block from the blocks tab
3. Observe that the block has been inserted into the post and that is in the proper position

</details>

<details><summary><strong>Inserter menu from Group block</strong></summary>

1. Add a Group block
2. Tap on the appender button (➕ button within the block)
3. Observe that the reusable tab is shown
4. Tap on a block
5. Tap on the appender button again (➕ button within the block)
6. Navigate to the reusable tab
7. Tap on a reusable block
8. Observe that the reusable block is inserted as an inner block

</details>

<details><summary><strong>Inserter menu from Social icons block</strong></summary>

1. Add a Social icons block
2. Tap on the appender button (➕ button within the block)
3. Observe that only allowed blocks are displayed
4. Tap on a block
5. Observe that the block is inserted as an inner block
6. 
</details>

<details><summary><strong>Preserve the scroll-position</strong></summary>

1. Tap on the ➕ button to display the inserter menu
2. Scroll down the content
3. Navigate to the reusable tab
4. Scroll down the content
5. Navigate to the blocks tab
6. Observe that the scroll on blocks tab is at the position as previously
7. Navigate to the reusable tab
8. Observe that the scroll on blocks tab is at the position as previously

</details>

<details><summary><strong>Landscape mode</strong></summary>

1. Tap on the ➕ button to display the inserter menu
2. Rotate the device to landscape mode
3. Observe that the inserter menu and the blocks tab are displayed properly
4. Navigate to the reusable tab
5. Observe that the reusable tab is displayed properly

</details>

<details><summary><strong>Swipe to close (bottom-sheet)</strong></summary>

1. Tap on the ➕ button to display the inserter menu
2. Scroll down the content
3. Scroll up to the top
4. Observe that the bottom-sheet can be closed by swiping down
5. Tap on the ➕ button to display the inserter menu
6. Navigate to the reusable tab
7. Scroll down the content 
8. Scroll up to the top
9. Observe that the bottom-sheet can be closed by swiping down

</details>

<details><summary><strong>Empty list of reusable blocks</strong></summary>

1. Remove all reusable blocks from the site (if they're trashed shouldn't be displayed)
2. Tap on the ➕ button to display the inserter menu
3. Observe that the reusable tab is not displayed

</details>

## Screenshots 

iOS|Android
--|--
<video src=https://user-images.githubusercontent.com/14905380/117045529-e602f400-ad0f-11eb-8f2c-2a1bce78dd68.mp4>|<video src=https://user-images.githubusercontent.com/14905380/117045557-ec916b80-ad0f-11eb-85f2-8a23acabcc9e.mp4>

## Types of changes

New feature

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/components/src/mobile/bottom-sheet/index.native.js
Comment (fluiddot) [inline packages/components/src/mobile/bottom-sheet/index.native.js]: This component only allowed a text as the header, this new prop `header` will allow us to add anything we might need to render in the header.

File: packages/block-editor/src/components/block-types-list/index.native.js
Comment (fluiddot) [inline packages/block-editor/src/components/block-types-list/index.native.js:25]: The logic for this component has been extracted from [`InserterSearchResults ` component](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/inserter/search-results.native.js). This component is now used for rendering the block types in the different tabs of the inserter menu as well as in the search results.

File: packages/block-editor/src/components/inserter/hooks/use-clipboard-block.native.js
Comment (fluiddot) [inline packages/block-editor/src/components/inserter/hooks/use-clipboard-block.native.js:13]: The logic for this hook has been extracted from the [`getItems` function of the inserter menu](https://github.com/WordPress/gutenberg/blob/ae7ab4528ed8afd17334ea054249af3a666e50ad/packages/block-editor/src/components/inserter/menu.native.js#L152-L173).

File: packages/block-editor/src/components/inserter/menu.native.js
Comment (fluiddot) [inline packages/block-editor/src/components/inserter/menu.native.js]: The inserter menu can be triggered by the ➕ button and within blocks like the social icons. Reusable blocks should only be shown in the first case (➕ button) so we have to check if `rootClientId` is undefined.

File: packages/block-editor/src/components/inserter/menu.native.js
Comment (fluiddot) [inline packages/block-editor/src/components/inserter/menu.native.js]: When the keyboard is shown/hide, it dispatches a layout animation and as we're also rendering different content when that happens (actually when the search form is active), these handlers are used to group all the layout changes in one single layout animation.

**Comment by fluiddot** [comment]
@jhnstn I'm about to set this [PR](https://github.com/WordPress/gutenberg/pull/30595) in ready for review but since I included your changes from this PR, I'm not sure if I should revert them, I'd appreciate your feedback on this, thanks 🙇 !

**Comment by jhnstn** [comment]
> @jhnstn I'm about to set this [PR](https://github.com/WordPress/gutenberg/pull/30595) in ready for review but since I included your changes from this PR, I'm not sure if I should revert them, I'd appreciate your feedback on this, thanks 🙇 !

Probably best to revert the changes. I'm not working on my PR this week and I'd like to get some of the changes you pulled in reviewed separately, thanks!

**Comment by fluiddot** [comment]
> Probably best to revert the changes. I'm not working on my PR this week and I'd like to get some of the changes you pulled in reviewed separately, thanks!

Ok, thanks for the feedback! I've just reverted it in this [commit](https://github.com/WordPress/gutenberg/pull/28495/commits/c8a19edc870e781a4a9440485055f735da2291b6) 👍 .

File: packages/components/src/mobile/inserter-button/index.native.js
Comment (fluiddot) [inline packages/components/src/mobile/inserter-button/index.native.js:93]: I set the maximum number of lines for the block's title to three, this will prevent rows to be too big but enough for display long texts.

Before|After
--|--
|

File: packages/components/src/mobile/bottom-sheet/index.native.js
Comment (fluiddot) [inline packages/components/src/mobile/bottom-sheet/index.native.js:302]: I found out an issue regarding this part when changing tabs. **I followed the following steps to reproduce it:**
1. Scroll down the blocks tab
1. Navigate to the reusable tab
1. Navigate back to the blocks tab
1. Observe that when scrolling up instead of scrolling the content the modal is being closed.

Curiously, the bounces value is used to control whether the modal should enable the swipe interaction that closes it so I had to modify it ([code reference](https://github.com/WordPress/gutenberg/blob/2a938548e8d2fed8a25b30db2b8675236089f176/packages/components/src/mobile/bottom-sheet/index.native.js#L410-L413)).

In any case, I think `bounces` should be enabled if it's not close to the top not only if it's closed to the bottom, I'm not sure what was the original purpose of this part but as far as I checked, this didn't introduce any regression.

File: packages/components/src/mobile/bottom-sheet/index.native.js
Comment (fluiddot) [inline packages/components/src/mobile/bottom-sheet/index.native.js]: `onHeaderLayout` is executed when the bottom-sheet component is mounted so if the component doesn't make any change in the layout, any other change made after it was closed would be animated. This doesn't particularly introduce a bug but I found out not consistent that some interactions were animated and others not.

File: packages/block-editor/src/components/inserter/tabs.native.js
Comment (fluiddot) [inline packages/block-editor/src/components/inserter/tabs.native.js:62]: The bottom-sheet component uses the `onScroll` handler to [calculate the value of `bounces`](https://github.com/WordPress/gutenberg/blob/2a938548e8d2fed8a25b30db2b8675236089f176/packages/components/src/mobile/bottom-sheet/index.native.js#L295-L301), this value is also used to control whether the modal should enable the swipe interaction that closes it, so it's important to explicitly call `onScroll` when the current tab changes to prevent undesired effects.

_Related to this [comment](https://github.com/WordPress/gutenberg/pull/28495/files#r627322782)._

**Comment by fluiddot** [comment]
~~⚠️ Until the reusable block is [enabled for all users by removing `devFlag`](https://github.com/WordPress/gutenberg/pull/31154), we shouldn't merge this PR.~~

**EDIT:** The reusable tab won't be shown in production until the reusable block is enabled so it's safe to merge this PR, we'll only see the tabs on development mode.

**Comment by guarani** [comment]
Hi @fluiddot, I took a look at this today but only managed a partial review so I'll have to continue this again later. Is this a feature branch with some already reviewed code, or is it all new code?

I haven't checked the code yet, I just tested on Android on all three types of sites and reusable blocks are working great for me in the inserter! 💯 I noticed that I had to log in again for the app to pick up a newly added reusable block, is that expected? I was on a self-hosted site (not connected via Jetpack) and tried reopening the editor, pulling-to-refresh the site list, etc. but it was only when I reinstalled the app (there's no logout option in the app for self-hosted sites), did the reusable block that I'd previously added appear.

I'll come back to this for an iOS test and code review.

**Comment by fluiddot** [comment]
Thank you very much @guarani for reviewing the PR ❤️ !

>Hi @fluiddot, I took a look at this today but only managed a partial review so I'll have to continue this again later. Is this a feature branch with some already reviewed code, or is it all new code?

The code of this PR is new, one thing I did to reduce its size was to extract some of the fixes I initially applied here to separate PRs but all of them are already in `trunk` so all the code here needs to be reviewed.

>I haven't checked the code yet, I just tested on Android on all three types of sites and reusable blocks are working great for me in the inserter! 💯 I noticed that I had to log in again for the app to pick up a newly added reusable block, is that expected? I was on a self-hosted site (not connected via Jetpack) and tried reopening the editor, pulling-to-refresh the site list, etc. but it was only when I reinstalled the app (there's no logout option in the app for self-hosted sites), did the reusable block that I'd previously added appear.

That's interesting, I'll try to reproduce it just in case it's a bug. Not sure if it's related, but on Android network requests made from RN are cached 10 minutes by default (I have a [PR](https://github.com/WordPress/gutenberg/pull/31186) to address this for reusable blocks), that could explain why you had to reinstall, although, this should happen on every site including WP.com 🤔 .

>I'll come back to this for an iOS test and code review.

Awesome! Let me know if you need help with the review.

**Comment by fluiddot** [comment]
>That's interesting, I'll try to reproduce it just in case it's a bug. Not sure if it's related, but on Android network requests made from RN are cached 10 minutes by default (I have a PR to address this for reusable blocks), that could explain why you had to reinstall, although, this should happen on every site including WP.com 🤔 .

I managed to reproduce it and I'm almost sure that it's produced by the network request caching on Android, actually, it happened on the three different types of sites. [The TTL of the cache is 10 minutes](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/30cab6f0045258b513e173c325b10dc25e19ee73/fluxc/src/main/java/org/wordpress/android/fluxc/network/BaseRequest.java#L36) so you should see the new reusable blocks after that time.

As this behavior is not ideal, my plan is to merge the [PR that fixes it](https://github.com/WordPress/gutenberg/pull/31186) before we enable the reusable block and merge this one.

File: packages/block-editor/src/components/block-types-list/style.native.scss
Comment (geriux) [inline packages/block-editor/src/components/block-types-list/style.native.scss]: Not to be picky but since we moved this, could we use the new naming? What do you think?

File: packages/block-editor/src/components/inserter/blocks-types-tab.native.js
Comment (geriux) [inline packages/block-editor/src/components/inserter/blocks-types-tab.native.js]: Could we rename this to `block-types-tab.native.js` so it matches its web version?

File: packages/block-editor/src/components/inserter/hooks/use-clipboard-block.native.js
Comment (geriux) [reply packages/block-editor/src/components/inserter/hooks/use-clipboard-block.native.js:13]: Since this is for mobile-only, what do you think if we put it under the same `components/src/mobile/clipboard` folder? I like that this was extracted 👍

File: packages/block-editor/src/components/inserter/menu.native.js
Comment (geriux) [reply packages/block-editor/src/components/inserter/menu.native.js]: Could we move these arrow functions?

File: packages/block-editor/src/components/inserter/search-results.native.js
Comment (geriux) [inline packages/block-editor/src/components/inserter/search-results.native.js:32]: This looks so clean now 😄

File: packages/block-editor/src/components/inserter/tabs.native.js
Comment (geriux) [inline packages/block-editor/src/components/inserter/tabs.native.js]: Could we move this style to a const?

File: packages/block-editor/src/components/block-types-list/style.native.scss
Comment (fluiddot) [reply packages/block-editor/src/components/block-types-list/style.native.scss]: Yeah, I totally agree, I used the BEM naming for the tabs component but I forgot to apply it here, thanks for pointing it out 🙇 !

File: packages/block-editor/src/components/inserter/blocks-types-tab.native.js
Comment (fluiddot) [reply packages/block-editor/src/components/inserter/blocks-types-tab.native.js]: Oops, good catch! I'll rename it, thanks!

File: packages/block-editor/src/components/inserter/tabs.native.js
Comment (fluiddot) [reply packages/block-editor/src/components/inserter/tabs.native.js]: Sure! I'm thinking that I might even memoize it to prevent potential re-renders, wdyt?

File: packages/block-editor/src/components/inserter/menu.native.js
Comment (fluiddot) [reply packages/block-editor/src/components/inserter/menu.native.js]: Sure! I'll create them using `useCallback` to prevent re-rendering 👍 .

File: packages/block-editor/src/components/inserter/hooks/use-clipboard-block.native.js
Comment (fluiddot) [reply packages/block-editor/src/components/inserter/hooks/use-clipboard-block.native.js:13]: Oh good point, I'll put it there since it could be used in other places, thanks!

Comment (fluiddot) [reply packages/block-editor/src/components/inserter/hooks/use-clipboard-block.native.js:13]: One thing I'm a bit concerned about regarding moving it to `packages/components` is that we would introduce a dependency into `components` package to `block-editor` which might not a good idea. I'm not sure how we usually deal with cross packages dependencies, wdyt?

File: packages/block-editor/src/components/block-types-list/style.native.scss
Comment (fluiddot) [reply packages/block-editor/src/components/block-types-list/style.native.scss]: I changed the naming in this [commit](https://github.com/WordPress/gutenberg/pull/28495/commits/958dd9d19ebb35d73cd2d23d4211f98f569f6c67).

File: packages/block-editor/src/components/inserter/blocks-types-tab.native.js
Comment (fluiddot) [reply packages/block-editor/src/components/inserter/blocks-types-tab.native.js]: I've renamed this component in this [commit](https://github.com/WordPress/gutenberg/pull/28495/commits/08f45490fbc8f16be4957c783d13bf18c2618e6e).

File: packages/block-editor/src/components/inserter/tabs.native.js
Comment (fluiddot) [reply packages/block-editor/src/components/inserter/tabs.native.js]: I moved these styles to a constant in this [commit](https://github.com/WordPress/gutenberg/pull/28495/commits/6153e92ad76c42ad40d89101b44ff0cafa731fc5), I finally decided not to memoize it because we don't have too many re-renders here, anyways if you think it would be worth it, I could apply it.

**Review by guarani — COMMENTED** [review]
> I managed to reproduce it and I'm almost sure that it's produced by the network request caching on Android, actually, it happened on the three different types of sites. The TTL of the cache is 10 minutes so you should see the new reusable blocks after that time.

> As this behavior is not ideal, my plan is to merge the PR that fixes it before we enable the reusable block and merge this one.

Sounds good 👍 Thanks for checking!

---

I came back to this today and tried it out on iOS, using the code from the Gutenberg Mobile PR, but after updating the submodule to point to the latest commit on this branch.

#### Changing the device orientation leads to layout issues

After opening the block picker and rotating to landscape orientation, the block picker shows both regular and reusable blocks, the content is cut-off, and the background dim on the right-side is missing. This same scenario on WPiOS `develop` doesn't seem to have this issue.

<details>
  <summary>See screenshots</summary>

| This branch | Current WPiOS develop |
| - | - |
| Screen Shot 2021-05-10 at 21 30 56 | Screen Shot 2021-05-10 at 21 34 52 |
  
</details>

#### Bottom sheet animates in from the left

The bottom sheet animates in from the bottom left whereas previously I think it animated in from bottom-center (recorded using Debug > Slow Animations in the simulator):

<details>
  <summary>See screenshots</summary>

| Screen Shot 2021-05-10 at 21 30 56 |
| - |  
</details>

#### Bottom sheet animation between tabs shows a blank area

The animation when switching between tabs looked slightly different to what I expected and it looks like there's an empty "page" between the first and second tab. I expected the content to be just off-screen rather than have a whole bottom sheet page between the regular block tab and the reusable block tab.

<details>
  <summary>See screenshots</summary>

| Screen Shot 2021-05-10 at 21 30 56 |
| - |  
</details>

---

I don't think the animation issues were present on Android when I checked last week, so they might be iOS-specific or due to recent changes. Also, I don't think the landscape issue was present on Android either.

I tested a WP.com and Atomic site, and everything is working well. If these issues I mention aren't blockers (e.g. they're not caused by changes in this PR), I'd be happy to have this merged.

File: packages/block-editor/src/components/inserter/tabs.native.js
Comment (geriux) [reply packages/block-editor/src/components/inserter/tabs.native.js]: I think it's ok to not memoize them 👍 Thanks for the change!

File: packages/block-editor/src/components/inserter/hooks/use-clipboard-block.native.js
Comment (geriux) [reply packages/block-editor/src/components/inserter/hooks/use-clipboard-block.native.js:13]: Oh! You're right, I think we can leave it here 😃

**Comment by fluiddot** [comment]
Thank you very much @guarani for the extensive review ❤️ !

I'm going to check all the issues you commented on and apply the potential fixes:

## Changing the device orientation leads to layout issues
>After opening the block picker and rotating to landscape orientation, the block picker shows both regular and reusable blocks, the content is cut-off, and the background dim on the right-side is missing. This same scenario on WPiOS develop doesn't seem to have this issue.

I managed to reproduce this issue in older commits from Gutenberg's `trunk` branch too so I'm not sure if this PR introduced it. However, looks like this only happens when testing in a simulator, I tried on a device and I couldn't reproduce it (check the following video recordings).

<details><summary><strong>Video recordings (commit used: e4429984fcb4d2791ae75319ec64569ec21c49c7)</strong></summary>

iPhone 11 simulator|iPhone 11 device
--|--
<video src=https://user-images.githubusercontent.com/14905380/117797025-e2abc300-b24f-11eb-8835-75337ece0f37.mp4 width=500>|<video src=https://user-images.githubusercontent.com/14905380/117797039-e6d7e080-b24f-11eb-8dbb-0e119e2fd47e.MP4 width=500>

</details>

## Bottom sheet animates in from the left
>The bottom sheet animates in from the bottom left whereas previously I think it animated in from bottom-center (recorded using Debug > Slow Animations in the simulator):

I couldn't manage to reproduce this issue, in my case enabling `Slow Animations in the simulator` it's not really slowing down the inserter menu animation (see attached video recording) 🤔 . I'd appreciate it if you could share the steps you followed to reproduce it just in case I'm overlooking anything.

<details><summary><strong>Video recording (iPad Pro 12.9-inch - 4th generation)</strong></summary>

https://user-images.githubusercontent.com/14905380/117799168-1d165f80-b252-11eb-8704-9a1235f87663.mp4

</details>

## Bottom sheet animation between tabs shows a blank area
>The animation when switching between tabs looked slightly different to what I expected and it looks like there's an empty "page" between the first and second tab. I expected the content to be just off-screen rather than have a whole bottom sheet page between the regular block tab and the reusable block tab.

Good catch! I realized that I was using the window width for positioning the tabs, which is wrong. I've updated the tabs and now I'm using the width of the wrapper so it shouldn't show a blank area, I added the change in this [commit](https://github.com/WordPress/gutenberg/pull/28495/commits/444386a2c1570a743339a09fa81c9ed678c5e540).

<details><summary><strong>Video recordings</strong></summary>

iPad Pro 12.9-inch - 4th generation|iPhone 11
--|--
<video src=https://user-images.githubusercontent.com/14905380/117801303-6ff11680-b254-11eb-94f1-f7cc050994e3.mp4 width=500>|<video src=https://user-images.githubusercontent.com/14905380/117801324-74b5ca80-b254-11eb-924a-64234e43518e.mp4 width=500>

</details>

<hr>

>I don't think the animation issues were present on Android when I checked last week, so they might be iOS-specific or due to recent changes. Also, I don't think the landscape issue was present on Android either.

Not sure, it could be caused by both actually although recent changes shouldn't have modified the behavior.

>I tested a WP.com and Atomic site, and everything is working well. If these issues I mention aren't blockers (e.g. they're not caused by changes in this PR), I'd be happy to have this merged.

I'd say that they aren't critical blockers but I'd like to have them solved if possible, the third issue was caused by this PR so that one is already fixed. The others look like weren't introduced in this PR, as far as I checked, anyways I'd appreciate it if you could review my feedback just in case I'm overlooking anything.

File: packages/block-editor/src/components/inserter/menu.native.js
Comment (fluiddot) [reply packages/block-editor/src/components/inserter/menu.native.js]: I made this change in this [commit](https://github.com/WordPress/gutenberg/pull/28495/commits/89a2e437d5004d78ed603d64711260a1581170b1).

**Review by geriux — APPROVED** [review]
LGTM! I tested the different scenarios between iOS and Android. Nice work 👏

**Review by guarani — APPROVED** [review]
> I managed to reproduce this issue in older commits from Gutenberg's trunk branch too so I'm not sure if this PR introduced it. However, looks like this only happens when testing in a simulator, I tried on a device and I couldn't reproduce it (check the following video recordings).

Thanks for checking, I tried again to reproduce but couldn't, so I think we can ignore this due to either being a simulator issue or an issue in `trunk` that was fixed.

> I couldn't manage to reproduce this issue, in my case enabling `Slow Animations in the simulator` it's not really slowing down the inserter menu animation (see attached video recording) 🤔 . I'd appreciate it if you could share the steps you followed to reproduce it just in case I'm overlooking anything.

I'm sorry to say I this isn't working for me anymore 😅, I'm not sure what I did to make slow animations work on the bottom sheet. 

> Good catch! I realized that I was using the window width for positioning the tabs, which is wrong. I've updated the tabs and now I'm using the width of the wrapper so it shouldn't show a blank area, I added the change in this [commit](https://github.com/WordPress/gutenberg/pull/28495/commits/444386a2c1570a743339a09fa81c9ed678c5e540).

Thanks! I confirmed now this looks good.

I think this can be merged now, I've done the test scenarios myself and @geriux has code-reviewed I believe.

I take it the `[DO NOT MERGE]` in the PR title is no longer needed, unless there's something pending, I think this looks ready to merge! Thanks a lot for getting this done ❤️

**Comment by fluiddot** [comment]
I originally set the PR as not ready to merge because of this [comment](https://github.com/WordPress/gutenberg/pull/28495#issuecomment-833523206):
>⚠️ Until the reusable block is enabled for all users by removing devFlag, we shouldn't merge this PR.

By the time I posted this comment, there was a blocker that I didn't know how much it would take to fix. Now the situation is different because the reusable block will only be available for WP.com sites and hopefully the PR that will enable them will be released soon as it's in review.

Being this said, I'll do a quick check just to assure that the reusable tab is not shown on self-hosted sites and if it works, I'll merge it.

Thank you very much @guarani and @geriux for the awesome review you did on this PR ❤️ !

**Comment by guarani** [comment]
Thanks for clarifying, @fluiddot. I'd missed that comment.

**Comment by fluiddot** [comment]
⚠️ I've found a small issue related to the bottom padding of the blocks list and the safe area. I'll push a quick fix once I solved it and let you know, sorry for the inconveniences 🙏 .

**Comment by fluiddot** [comment]
> ⚠️ I've found a small issue related to the bottom padding of the blocks list and the safe area. I'll push a quick fix once I solved it and let you know, sorry for the inconveniences 🙏 .

Here is the [commit](https://github.com/WordPress/gutenberg/pull/28495/commits/00289fce04e794ac79495e91b1550cdc2efc37d8) where I added the fix. @geriux @guarani let me know if you if you approve the change and if so I'll merge the PR, thanks 🙇 .

**Comment by guarani** [comment]
> > ⚠️ I've found a small issue related to the bottom padding of the blocks list and the safe area. I'll push a quick fix once I solved it and let you know, sorry for the inconveniences 🙏 .
> 
> Here is the [commit](https://github.com/WordPress/gutenberg/pull/28495/commits/00289fce04e794ac79495e91b1550cdc2efc37d8) where I added the fix. @geriux @guarani let me know if you if you approve the change and if so I'll merge the PR, thanks 🙇 .

Looks good to me, I tested on iOS and the bottom sheet spacing appeared OK (although I wasn't sure what to look out for).
I'd say it's good to go :shipit:

**Comment by fluiddot** [comment]
> > > ⚠️ I've found a small issue related to the bottom padding of the blocks list and the safe area. I'll push a quick fix once I solved it and let you know, sorry for the inconveniences 🙏 .
> > 
> > 
> > Here is the [commit](https://github.com/WordPress/gutenberg/pull/28495/commits/00289fce04e794ac79495e91b1550cdc2efc37d8) where I added the fix. @geriux @guarani let me know if you if you approve the change and if so I'll merge the PR, thanks 🙇 .
> 
> Looks good to me, I tested on iOS and the bottom sheet spacing appeared OK (although I wasn't sure what to look out for).
> I'd say it's good to go :shipit:

True, I forgot to add details about the issue, here is a screenshot related to this:



As you can see, the content is too close to the bottom area, that's because I wasn't including the safe area, now it looks this way:

**Comment by geriux** [comment]
> Here is the [commit](https://github.com/WordPress/gutenberg/pull/28495/commits/00289fce04e794ac79495e91b1550cdc2efc37d8) where I added the fix. @geriux @guarani let me know if you if you approve the change and if so I'll merge the PR, thanks 🙇 .

LGTM!

**Comment by fluiddot** [comment]
@geriux @guarani I also assigned you as reviewers of the [GB-mobile PR](https://github.com/wordpress-mobile/gutenberg-mobile/pull/3054) associated with this one, when you're available I'd appreciate it if you could approve it too, thanks 🙇 !

**Pull request merged**
